### PR TITLE
Add Docker deployment method to QuickStart guide for Version 4.x

### PR DESCRIPTION
**Pull request by kaxiya1021**
## What is the purpose of the change

The purpose of this change is to enhance the QuickStart guide by adding a Docker deployment method specifically tailored for RocketMQ Version 4.x.

### CODE REVIEW

File: docs/01-introduction/04quickstartWithDocker4.x.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ SendResult [sendStatus=SEND_OK, msgId= ...
+
+$ sh tools.sh org.apache.rocketmq.example.quickstart.Consumer
+ ConsumeMessageThread_%d Receive New Messages: [MessageExt...
+```
+
+
+## 6.SDK测试消息收发
+
+工具测试完成后，我们可以尝试使用 SDK 收发消息。这里以 Java SDK 为例介绍一下消息收发过程，可以从 [rocketmq-clients](https://github.com/apache/rocketmq-clients) 中参阅更多细节。
```
Comment (RongtongJin) [inline docs/01-introduction/04quickstartWithDocker4.x.md]: 4.x用的不是rocketmq-clients的仓库哈，用的主仓库的client模块。

File: docs/01-introduction/04quickstartWithDocker4.x.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+               System.out.println("Message sending failed.");
+           } finally {
+               // 关闭生产者
+               producer.shutdown();
+           }
+       }
+   }
+   ```
+
+4. 在已创建的Java工程中，创建订阅普通消息程序并运行。Apache RocketMQ 支持[SimpleConsumer](https://rocketmq.apache.org/zh/docs/featureBehavior/06consumertype)和[PushConsumer](https://rocketmq.apache.org/zh/docs/featureBehavior/06consumertype)两种消费者类型，您可以选择以下任意一种方式订阅消息。
```
Comment (RongtongJin) [inline docs/01-introduction/04quickstartWithDocker4.x.md]: 4.x没有SimpleConsumer

File: docs/01-introduction/04quickstartWithDocker4.x.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+deleteWhen = 04
+# 以小时计算的文件保留时间 默认值72小时
+fileReservedTime = 48
+# Broker角色
+brokerRole = ASYNC_MASTER
+# 刷盘方式
+flushDiskType = ASYNC_FLUSH
+# 此处为示例，实际使用时请替换为真实的 Broker 地址
+brokerIP1 = 127.0.0.1
+```
```
Comment (RongtongJin) [inline docs/01-introduction/04quickstartWithDocker4.x.md]: 能否用默认的broker.conf而不是让客户自己再去编写

**Review by ShannonDing — APPROVED** [review]
+1, LGTM

**Review by RongtongJin — APPROVED** [review]

**Comment by ShannonDing** [comment]
it also works well on macOS.

**Pull request merged**